test/viewer.js

```diff
@@ -0,0 +1,50 @@
+const net = require('net');
+
+const Logger = require('../lib/Logger');
+const { startServer } = require('../lib/viewer.js');
+
+describe('WebSocket server', function () {
+  it('should not crash when an error is emitted on the websocket', function (done) {
+    const bundleStats = {
+      assets: [{ name: 'bundle.js', chunks: [0] }]
+    };
+
+    const options = {
+      openBrowser: false,
+      logger: new Logger('silent'),
+      port: 0
+    };
+
+    startServer(bundleStats, options).then(function ({ http: server }) {
+      const socket = net.createConnection(server.address().port, function () {
+        socket.write([
+          'GET / HTTP/1.1',
+          'Host: localhost',
+          'Upgrade: websocket',
+          'Connection: Upgrade',
+          'Sec-WebSocket-Key: Ix02x1E8kmC9ie01/dGCuw==',
+          'Sec-WebSocket-Version: 13',
+          '\r\n'
```

Comment:
Do we need an additional `\r\n` here or would an empty string be enough? I see that you're joining the resulting array with `\r\n` anyway, so this seems a bit excessive.

If we indeed want to have two empty lines, could we instead have two empty strings in the array before joining?

Reply:
Each header line must end with `\r\n` and we need an empty one ending `\r\n` to signal the end.